### PR TITLE
Clean back bay with less than 5 people

### LINKED ISSUE #54925: Back Bay Limbo
### Describe the bug

Borichenko gives you a mission to recruit 5 people to help w/ cleaning the back bay of the refugee center after you complete the mission to clear it of Z's. It's possible to fail your Social skill checks enough times that you will find yourself unable to complete the mission. This effectively blocks access to his quest chain.

### Steps to reproduce

1. Clear out the back bay.
2. Talk to Borichenko.
3. Fail to convince any1 else to help clean.
4. Pay 10 Merch to recruit 2 people from downstairs.
5. Realize you were scammed.

### Expected behavior

I expected there to be a failure condition for the mission. As it stands, I'm out 10 Merch & the mission is permanently stuck in the active missions section.

### Screenshots

_No response_

### Versions and configuration

Android 0.F-3 Tiles English vanilla

### Additional context

Would be nice if there were more options to buy your way to success using Merch. Alternatively, being able to leverage the good will you've gained from doing other missions in the center could help. What about dialogue options for recruiting the beggars to help??

### LINKED ISSUE #65185: Mission "Find someone to clean up the back bay" breaks if all social checks are failed
### Describe the bug

Boris's mission to find five people to clean up the back bay requires you to get the help of five people total. The teamster will work without persuasion, Smokes will give you either one worker or three depending on how much you pay him, and several others can be persuaded. When you reach five, the quest's done.

However, once you've asked someone you can persuade and they've said no, or if you've asked Smokes and paid him for one worker, you can't ask again. And the only guaranteed worker is the teamster. Meaning that if your persuasion skills aren't up to snuff, or you haven't ingratiated yourself to the people you can persuade, you can fail ALL the persuasion checks and end up with two to four total workers, making the quest impossible to complete.

### Attach save file

n/a

### Steps to reproduce

1. Create character with zero social and disadvantages (just to be sure; I encountered this bug with 6 social and no advantages or disadvantages socially).
2. Find the refugee center.
3. Complete the "clear the back bay" mission.
4. Take the "find people" mission from Boris.
5. Save game to savescum.
6. Ask everybody who can be asked, repeating this until everyone says no.
7. Observe that the quest is uncompletable.

### Expected behavior

Possibilities to solve:
--Option to help clean the bay yourself, adding one to the list (this doesn't solve it by itself, because you can still get stuck if you only buy one helper off Smokes)
--Option to assign NPC companions to help (you might need to find three of them, which is still a bit much, but doable)
--Option to ask other survivors again, after a period of time has passed
--Option to pay Smokes for more than just three workers

Paying Smokes is probably the easiest to code. Have him keep the 5 Merch option to hire another worker until the quest is complete (or until you have five), and you can hire as many workers as you like.

### Screenshots

_No response_

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19044.2846 (21H2)
- Game Version: 5b32e27 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    Blaze Industries [blazeindustries],
    Magiclysm [magiclysm],
    Xedra Evolved [xedra_evolved],
    Generic Guns [generic_guns],
    DinoMod [DinoMod],
    Megafauna [megafauna],
    Extra Mutated Scenarios [extra_mut_scens],
    Mining Mod [Mining_Mod],
    More Survivor Stuff [more_survivor_stuff],
    Mythical Martial Arts [MMA],
    Translate Complex Dialogue [translate_dialogue],
    Magiclysm No Class Limit [magiclysm_no_class_limit],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills],
    Tamable Wildlife [Tamable_Wildlife]
]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by anoobindisguise**
#### Summary
content "alternative resolution to clean back bay quest"

#### Purpose of change
Fixes #54925
Fixes #65185
There were only 4 guaranteed people to clean up the back bay. There were 3-4 people you could do a social check for who may all refuse. This lead to the whole refugee center deciding to just leave the back bay filthy and uncleaned despite having as many as 4 people willing to work on it which felt really silly.

#### Describe the solution
Really simple solution, but I'm new to dialogue. This may be too hacky though.
Lets you go back to Boris and tell him you you couldn't find enough people and offer to help yourself.
If you only have 2 people, Boris agrees to help too and you spend 5 hours standing in front of Boris pretending to clean, and then you can talk to him again to complete the quest as it increments the variable of people who agreed to help by 3.
If you have 3 people same as above, but you only spend 4 hours.
If you have 4 people, Boris agrees to help but decides that 4 is enough and they don't need your help cleaning. Then speak to him again as usual to complete the quest.

#### Describe alternatives you've considered
Rework the quest to not use the hardcoded "Mission success! I don't know what else to say." quest structure so that you can complete the quest immediately upon agreeing to help clean instead of having only 4 people left.
Apply a morale debuff for cleaning all the gore since that's presumably why the refugees don't wanna do it. I dislike this option, the player is built different and I kind of like it that way.
Add other options like recruiting the beggars, paying Makayla, paying the refugees, giving Smokes 15 merch to get 4 people + the teamster... 

#### Testing
Tested and confirmed functional.

### CODE REVIEW

**Comment by bombasticSlacks** [comment]
I would make the new counter use math instead of old school EOC but this looks like it works, the activity is also good practice yeah :)

File: data/json/npcs/refugee_center/surface_refugees/NPC_Boris_Borichenko.json [hunk 1]
```diff
@@ -422,6 +478,44 @@
       }
     ]
   },
+  {
+    "type": "talk_topic",
+    "id": "TALK_REFUGEE_Boris_MakeDoWith2",
+    "dynamic_line": "Hmm, well it is not ideal but I appreciate you trying and being willing to help.  I will have to do my best despite the bad memories.  Thank you, my friend.",
```
Comment (MNG-cataclysm) [inline data/json/npcs/refugee_center/surface_refugees/NPC_Boris_Borichenko.json:484]:
```suggestion
    "dynamic_line": "Well, it's not ideal, but I appreciate you trying and being willing to help.  I'll have to do my best despite the bad memories.  Thank you, my friend.",
```

File: data/json/npcs/refugee_center/surface_refugees/NPC_Boris_Borichenko.json [hunk 2]
```diff
@@ -422,6 +478,44 @@
...
+        { "u_adjust_var": "Boris_mission_1", "type": "mission", "context": "cleanup_promises", "adjustment": 3 },
+        { "u_assign_activity": "ACT_GENERIC_EOC", "duration": "300 minutes" },
+        { "npc_add_var": "cleanup_asked", "type": "mission", "context": "Boris_mission_1", "value": "yes" }
+      ]
+    }
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_REFUGEE_Boris_MakeDoWith3",
+    "dynamic_line": "Thank you for trying, my friend.  With you and me that will be five, I will do my best to be strong and assist in the cleanup.",
```
Comment (MNG-cataclysm) [inline data/json/npcs/refugee_center/surface_refugees/NPC_Boris_Borichenko.json:497]:
```suggestion
    "dynamic_line": "Thanks for trying, my friend.  With you and me that will be five, I'll do my best  and assist in the cleanup.",
```

File: data/json/npcs/refugee_center/surface_refugees/NPC_Boris_Borichenko.json [hunk 3]
```diff
@@ -422,6 +478,44 @@
...
+        { "u_adjust_var": "Boris_mission_1", "type": "mission", "context": "cleanup_promises", "adjustment": 2 },
+        { "u_assign_activity": "ACT_GENERIC_EOC", "duration": "240 minutes" },
+        { "npc_add_var": "cleanup_asked", "type": "mission", "context": "Boris_mission_1", "value": "yes" }
+      ]
+    }
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_REFUGEE_Boris_MakeDoWith4",
+    "dynamic_line": "That is quite alright, four should be enough though I appreciate your offer.  Four people will make the job much easier and I will do my best even if it is painful.  My thanks, friend.",
```
Comment (MNG-cataclysm) [inline data/json/npcs/refugee_center/surface_refugees/NPC_Boris_Borichenko.json:510]:
```suggestion
    "dynamic_line": "That's quite alright, four should be enough.  However, I appreciate your offer.  Four people will make the job much easier and I'll do my best, even if it brings up painful memories.  My thanks.",
```

**Comment by fairyarmadillo** [comment]
@MNG-cataclysm Boris does not use contractions because he is speaking with a heavy Russian accent.

**Pull request merged**